util/fpga/cw_spiflash.py

```diff
@@ -193,24 +206,28 @@ def __init__(self, fpga, board='CW305'):
         self._fpga_io.spi1_enable(True)
 
     def _reset_opentitan(self):
-        self._fpga_io.pin_set_state(self._PIN_MAPPINGS[self._board].PIN_JTAG_SPI, 1)
+        self._fpga_io.pin_set_state(self._PIN_MAPPINGS[self._board].PIN_TAP_STRAP1, 1)
```

Comment:
should this one be 0 also? Otherwise based on the current mapping, does this conflict with the SPI device pins?

Reply:
Hm, we have to check whether we can disentangle the SPI and JTAG. 
If I understand correctly, this has been set to set to 1 so that the JTAG for the RV_DM is selected by default.
SPI is then only selected during bootstrapping (see `__enter__` and `__exit__` functions below).

Reply:
i'm not sure this part matters actually ...because.. the changes necessary to enable jtag / openocd on cw310...was not merged :)
I think we should use this opportunity to disentangle (and maybe sync up with Colin again since they did the work for an FTDI emulation).

Reply:
Yeah we should disentangle this... there may be a way since slow interfaces like JTAG can be emulated via SAM3X GPIOs (as @a-will points out above).

Let's do that in a follow up to this so that we can at least get the strap pins into the right locations to unblock work on bootstrap.

Reply:
i'm kind of confused about this part.
If TAP_STRAP is set to 1, that means we're not in functional mode right? Then assuming the life cycle matches, since in the fpga pinmux we've set the SPI pins to JTAG, doesn't that mean we won't be able to boot-strap at all? Since the JTAG pins will be driving the spi output to sam3x? let me know what i missed.

Reply:
I guess this currently works since we run in a DFT enabled lifecycle state at the moment, which means that you can change the TAP mux selection at any time.

once we move to a prod/dev lifecycle state on FPGA, what you are pointing out could indeed become an issue since this pin is only sampled once at the beginning of the boot process and that would mean that the SPI would never be functional in that case.

If we do not plan to implement any life cycle TAP tests via SAM3X, this problem should however only occur during DEV, right?

Reply:
ah i just noticed that upon "enter" it is set to 0.
I think there's no immediate need we have to fix this now. When we disentangle the spi/jtag pins, we could just fix it altogether. Then it'll probably become part of the ftdi openocd setup file to set those pins or something.

Reply:
exactly! and that only works because we run in RMA (I believe) at the moment.

ok sounds good, let's do it that way.